app/api/internal/applications.rb

```diff
@@ -18,17 +18,25 @@ module API
           end
           app_attrs[:service_id] = service_id
           app_attrs[:id] = id
-          app = Application.save(app_attrs)
-          [201, headers, { status: :created, application: app.to_hash }.to_json]
+          begin
+            app = Application.save(app_attrs)
+            [201, headers, { status: :created, application: app.to_hash }.to_json]
+          rescue ApplicationHasNoState
+            [400, headers, { status: :bad_request, error: 'application cannot be created, it has no state' }.to_json]
```

Comment:
this does not look bad, but the caught exception could already encode the `bad_request` status and the error message and use that here, which would help in catching several errors at once - perhaps it's already encoding that?

Reply:
I have set ApplicationHasNoState as a BadRequest type of exception and then I obtain the http status code and the message from the exception object. I have pushed the changes

Reply:
Finally, after discussing it we do not set the http status code from the received exception because the received exception is from the Model and we do not want to rely on the http status code returned by the model in case it changes